lib/node-token-validation/src/api/TokenValidator.ts

```diff
@@ -49,6 +50,136 @@ export class TokenValidator {
         this.openIdConfigProvider = new OpenIdConfigProvider(this.config, this.networkInterface, this.logger);
     }
 
+    /**
+     * Middleware-style function to validate token from request.
+     * @param {@link (TokenValidationParameters:type)}
+     * @param resource Optional resource to retrieve access token from session 
+     * @returns 
+     */
+    validateTokenMiddleware(options: TokenValidationParameters, resource?: string): Function {
+
+        // @ts-ignore 
+        return (req: ExpressRequest, res: ExpressResponse, next: ExpressNextFunction) => {
+
+            // Adding access token to authorization header from session
+            if (resource && req.session && req.session.protectedResources) {
+                if (!req.session.protectedResources[resource]) {
+                    const error = ValidationConfigurationError.createMissingTokenError(`Resource ${resource} is not a protectedResource in req session. No access token to add to headers.`);
+                    next(error);
+                } else {
+                    const token = req.session.protectedResources[resource].accessToken;
+                    if (!token) {
+                        const error = ValidationConfigurationError.createMissingTokenError(`No access token for resource ${resource} in session. Unable to add to request headers.`);
+                        next(error);
```

Comment:
The assumption is that if someone is passing an (optional) resource into the `validateTokenMiddleware` API, that they're wanting an access token retrieved and validated for that resource. It's currently throwing an error when an access token is not found, but I suppose if someone was using this API on every call there could be times when an access token is not found and they would go fetch one instead. Unsure whether it would still be best to return an error that they would then handle, or pass something else along.

Reply:
An error here is fine for now IMO, but we could consider making it an actionable error like InteractionRequired?